### DISCUSSION TITLE
Question About Rules and Expectations

### DISCUSSION THREAD

**Discussion by nocholasrift** [utiasDSL/safe-control-gym#101] [IROS 2022 Competition]
Greetings! Thank you for organizing this competition, looking forward to it at IROS this year. I have a quick question regarding the allowable approaches that can be used. Are we allowed to use other machine learning approaches like supervised / imitation learning to accomplish this task, or is only reinforcement learning allowed?

Thanks again!

**Comment by JacopoPan** [comment]
Hello @nocholasrift 

Thanks for the question! The main "constraint/expectation" is to use the interface defined in class `Controller`/file `competition/edit_this.py` (so that `getting_started.py` can run without changes). There is no limitation on the type of approach that you can use. 

In fact, we welcome the proposal of methodologies for ALL areas of control, robotics, machine learning, and AI: one of the main objectives of this repository is to support *large spectrum* comparisons and benchmarks.

**Reply by nocholasrift** [reply]
Great, thank you Jacopo!
